### PR TITLE
Map color mode preference

### PR DESCRIPTION
**Pull request by AntonKhorev**
Includes #5339. Implements step 5 from https://github.com/openstreetmap/openstreetmap-website/issues/5324#issuecomment-2504102012.

image
image

### CODE REVIEW

File: app/assets/stylesheets/common.scss [hunk 1]
```diff
@@ -516,6 +511,23 @@ body.small-nav {
   }
 }
 
+@mixin dark-map-color-scheme {
+  .leaflet-tile-container .leaflet-tile,
+  .mapkey-table-entry td:first-child > * {
+    filter: brightness(.8);
```
Comment (hlfan) [inline app/assets/stylesheets/common.scss:517]: Doesn't the mentioned step 5 include the need for the filters to be set individually to give the cartographers a choice?
I don't see that as a necessity for this PR, but already setting the filter through a custom property defined on the layer could ease the transition to that being implemented and also act as default and fallback:
```css
here {
    filter: var(--dark-map-filter);
}
somewhere else {
    .leaflet-layer {
        --dark-map-filter: brightness(.8);
    }
}
```

Comment (AntonKhorev) [reply app/assets/stylesheets/common.scss:517]: Step 5 has an impossible requirement of "make it dark but don't pick any filter". The closest possible thing is "don't touch existing filters".

**Comment by omcnoe** [comment]
This will only be configurable for logged in users, meaning that non-logged in will still be stuck on the dark UI + dimmed tiles.

**Comment by gravitystorm** [comment]
Thanks @AntonKhorev , I think this works well.

**Pull request merged**